src/control/server/instance.go

```diff
@@ -69,9 +72,8 @@ type EngineInstance struct {
 	_lastErr    error // populated when harness receives signal
 }
 
-// NewEngineInstance returns an *EngineInstance initialized with
-// its dependencies.
-func NewEngineInstance(l logging.Logger, p *storage.Provider, jf systemJoinFn, r EngineRunner) *EngineInstance {
+// NewEngineInstance returns an *EngineInstance initialized with its dependencies.
+func NewEngineInstance(l logging.Logger, p *storage.Provider, jf systemJoinFn, r EngineRunner, ps *events.PubSub) *EngineInstance {
```

Comment:
I don't love the ripple effects of this change, but that is probably more of a criticism of the test code than anything else. I don't see a better way of injecting this dependency.

Reply:
Neither do I, I was thinking about making it a vararg so it didn't need to be included in all usages but that isn't great either.